### PR TITLE
Implementation changes for CiscoBgp4MIB

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
**- What I did**
Currently the data required for CiscoBgp4MIB is retrieved from bgpd deamon. snmp_ax_impl connects to bgpd daemon via tcp socket and retreives the BGP neighbor information required for CiscoBgp4MIB. This design is modified to support multi-asic platform. The data required by CiscoBgp4MIB can be populated in STATE_DB by a new daemon in BGP docker. 
This PR captures changes required in snmp_ax_impl to get data from STATE_DB.
This PR has dependency on the below PRs:
https://github.com/Azure/sonic-buildimage/pull/5329
https://github.com/Azure/sonic-buildimage/pull/5436
**- How I did it**
Changes made so that snmp_ax_impl can talk to STATE_DB and retrieve NEIGH_STATE_TABLE. This change will affect both single and multi asic platforms. In case of multi-asic platforms, snmp_ax_impl will retrive data from STATE_DB of all namespaces.

**- How to verify it**
Bring up single asic platform and multi-asic platforms with the changes, the output of iso.3.6.1.4.1.9.9.187.
Verify if the output matches with the information in STATE_DB NEIGH_STATE_TABLE.

**- Description for the changelog**

### CODE REVIEW

File: src/sonic_ax_impl/mibs/vendor/cisco/bgp4.py [hunk 1]
```diff
@@ -1,54 +1,61 @@
...
-        try:
-            if not self.sock.connected:
-                return
-
-            sessions = self.QuaggaClient.union_bgp_sessions()
+        for neigh_key, db_index in self.neigh_state_map.items():
+            neigh_str = neigh_key.decode()
+            neigh_str = neigh_str.split('|')[1]
+            neigh_info = self.db_conn[db_index].get_all(mibs.STATE_DB, neigh_key, blocking=False)
+            if neigh_info is None:
```
Comment (qiluo-msft) [inline src/sonic_ax_impl/mibs/vendor/cisco/bgp4.py:41]:
```suggestion
            if neigh_info:
```

File: tests/test_bgp.py [hunk 2]
```diff
@@ -18,8 +17,7 @@
 from ax_interface.constants import PduTypes
 from sonic_ax_impl.mibs.ietf import rfc4363
 from sonic_ax_impl.main import SonicMIB
-from sonic_ax_impl.lib.quaggaclient import parse_bgp_summary
-from mock_tables.socket import MockGetHostname
+from sonic_ax_impl.mibs.vendor.cisco.bgp4 import CiscoBgp4MIB
```
Comment (qiluo-msft) [inline tests/test_bgp.py:20]: >CiscoBgp4MIB [](start = 49, length = 12)

Not used?

File: src/sonic_ax_impl/mibs/vendor/cisco/bgp4.py (hunk 1 above)
Comment (SuvarnaMeenakshi) [reply src/sonic_ax_impl/mibs/vendor/cisco/bgp4.py]: Update the condition check/

File: tests/test_bgp.py (hunk 2 above)
Comment (SuvarnaMeenakshi) [reply tests/test_bgp.py:20]: quaggaclient and mock tables are not required anymore.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**